### PR TITLE
Adding Qwen3-Embedding Model support

### PR DESCRIPTION
**Pull request by arginugaTT**
### Ticket
Qwen3-Embedding Model support

### Problem description
Supporting Qwen3-Embedding model to work with tt-vllm-plugin of inference-server

### What's changed
1. Added generator_vllm.py file that interacts with tt-vllm-plugin to serve the model
2. Added model config entries for Qwen-embedding model
3. Added enums to identify embedding models (qwen3 & Bge), which will be used in tt-vllm-plugin

### Checklist

- [x] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/21195618195)
- [x] [Single Card Demo Tests](https://github.com/tenstorrent/tt-metal/actions/runs/21195562061)
- [ ] [T3K Demo Tests](https://github.com/tenstorrent/tt-metal/actions/runs/21195574676)
- [ ] [Galaxy Tests](https://github.com/tenstorrent/tt-metal/actions/runs/21195586500)

### CODE REVIEW

**Review by mbahnasTT — APPROVED** [review]

**Comment by yieldthought** [comment]
Codex review:

The new Qwen3 embedding wrapper currently returns LM-head logits (and for shorter prompts, logits for padding tokens), so embedding outputs are incorrect for typical usage. These correctness issues make the patch unreliable in its current form.

Full review comments:

- [P1] Use hidden states instead of logits for embeddings — [models/demos/wormhole/qwen3_embedding_8b/demo/generator_vllm.py:384](https://github.com/tenstorrent/tt-metal/blob/6d8952bb295a22e0c534f61eca85dabc9c1d6e42/models/demos/wormhole/qwen3_embedding_8b/demo/generator_vllm.py#L384)-386
  `Generator.prefill_forward_text` returns LM-head logits for the last token (shape `[B, 1, vocab_size]`), so assigning `embeddings = logits` causes the embedding endpoint to return vocab-sized logits rather than hidden-size embeddings. For any Qwen3-Embedding request this produces incorrect outputs unless the Transformer is modified to expose hidden states; consider extracting/pooling hidden states before the LM head or adding a model path that returns embeddings.

- [P2] Pass real prompt lengths to prefill — [models/demos/wormhole/qwen3_embedding_8b/demo/generator_vllm.py:362](https://github.com/tenstorrent/tt-metal/blob/6d8952bb295a22e0c534f61eca85dabc9c1d6e42/models/demos/wormhole/qwen3_embedding_8b/demo/generator_vllm.py#L362)-367
  `prefill_forward_text` uses `prompt_lens` to set `last_token_idx = seq_len - 1`; by passing the padded `seq_len` (max_seq_len), any input shorter than max_seq_len will return the padding token’s logits rather than the last real token, so embeddings become unrelated to the prompt. Use `original_seq_len` for `prompt_lens` (padding only for trace reuse) to avoid this for shorter sequences.

**Review by yieldthought — CHANGES_REQUESTED** [review]
I think codex is right: https://huggingface.co/Qwen/Qwen3-Embedding-8B#transformers-usage shows them using the last hidden states for the embedding, not the lm head output.

**Comment by arginugaTT** [comment]
> I think codex is right: https://huggingface.co/Qwen/Qwen3-Embedding-8B#transformers-usage shows them using the last hidden states for the embedding, not the lm head output.

I fixed both issues

**P1: Use hidden states instead of logits**

In `generator_vllm.py`:

- `return_hidden_states=True` is passed to `prefill_forward_text`
- `embeddings = hidden_states` (not logits)

Supporting changes:

- Added `process_hidden_states_after_prefill_trace()` in `model.py`  — extracts hidden states after norm, before LM head
- Added `process_output_prefill_hidden_states()` in `model.py`  — processes hidden states from host tensor
- Modified `prefill_forward_text()` in `generator.py`  — added `return_hidden_states` parameter

**P2: Pass real prompt lengths**

In `generator_vllm.py` :
- `prompt_lens=[original_seq_len] * batch_size`— uses `original_seq_len`, not `padded seq_len`

File: models/tt_transformers/tt/generator.py [hunk 1]
```diff
@@ -96,11 +96,6 @@ def __init__(self, model, model_args, mesh_device, processor=None, tokenizer=Non
         self.enable_split_sampling = True
         self.mode = None
 
-    # Class-level capabilities (VLLM specific, to be overridden by subclasses)
-    model_capabilities = {
-        "supports_prefix_caching": True,
-    }
-
```
Comment (skhorasganiTT) [inline models/tt_transformers/tt/generator.py:103]: You are undoing changes in this file that went in recently, please fix your rebase/branch

**Review by skhorasganiTT — CHANGES_REQUESTED** [review]

File: models/tt_transformers/tt/generator.py (hunk 1 above)
Comment (arginugaTT) [reply models/tt_transformers/tt/generator.py:103]: Hi @skhorasganiTT , I rebased the file properly.

**Review by yieldthought — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]
Please also test vLLM nightly - https://github.com/tenstorrent/tt-metal/actions/workflows/vllm-nightly-tests.yaml
